### PR TITLE
feat:impl in java and spring boot

### PR DESCRIPTION
**Pull request by mehdiqanbarzadeh**

### CODE REVIEW

**Review by geniuscodes — COMMENTED** [review]
dont have to push target folder

File: shipmentNotifier/src/main/java/io/billie/shipmentnotifier/api/shipment/ShipmentResource.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+
+	private final ShipmentService shipmentService;
+
+	private final ShipmentResourceMapper shipmentResourceMapper;
+
+
+	@PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE,
+			produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<ShipmentResponse> notifyShipment(@RequestBody @Valid ShipmentRequest request) throws BusinessException {
```
Comment (geniuscodes) [inline shipmentNotifier/src/main/java/io/billie/shipmentnotifier/api/shipment/ShipmentResource.java:36]: final

File: shipmentNotifier/src/main/java/io/billie/shipmentnotifier/exception/CustomizedResponseEntityExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	public final ResponseEntity<ResponseService> handleUnsupportedOperationException(UnsupportedOperationException ex,
+			WebRequest request) {
+		HttpHeaders headers = new HttpHeaders();
+		headers.setContentType(MediaType.APPLICATION_JSON);
+		return new ResponseEntity<>(new GeneralResponse(ResultStatus.FORBIDDEN_REQUEST), headers,
+				HttpStatus.UNPROCESSABLE_ENTITY);
+	}
+
+//	@ExceptionHandler(Exception.class)
+//	public final ResponseEntity<ResponseService> handleGeneralException(Exception ex, WebRequest request) {
```
Comment (geniuscodes) [inline shipmentNotifier/src/main/java/io/billie/shipmentnotifier/exception/CustomizedResponseEntityExceptionHandler.java:77]: still needed ?

File: shipmentNotifier/src/main/java/io/billie/shipmentnotifier/service/shipment/impl/ShipmentServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+	private final OrderService orderService;
+
+	private final ShipmentServiceMapper shipmentServiceMapper;
+
+	@Transactional
+	@Override
+	public Shipment processShipment(ShipmentDto shipmentDto) throws BusinessException {
+		log.info("Processing shipment: {}", shipmentDto);
+		Order order = orderService.findByTrackingCodeAndMerchantNumber(shipmentDto.getOrderTrackingCode(), shipmentDto.getMerchantNumber());
```
Comment (geniuscodes) [inline shipmentNotifier/src/main/java/io/billie/shipmentnotifier/service/shipment/impl/ShipmentServiceImpl.java:34]: final

File: shipmentNotifier/src/main/java/io/billie/shipmentnotifier/service/shipment/mapper/ShipmentServiceMapper.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+	@Named("toShipmentStatus")
+	default ShipmentStatus toShipmentStatus(boolean isSuccessful) {
+		return isSuccessful ? ShipmentStatus.SUCCESSFUL : ShipmentStatus.FAILED;
+	}
+
+	@Named("toShipmentDescription")
+	default String toShipmentDescription(boolean isSuccessful) {
+		return isSuccessful ? ResultStatus.SUCCESS.name() : ResultStatus.ShipmentAmountIsGreaterThatOrderAmountException.name();
+		// it's not best practise solution, it's all about time I don't have time
```
Comment (geniuscodes) [inline shipmentNotifier/src/main/java/io/billie/shipmentnotifier/service/shipment/mapper/ShipmentServiceMapper.java:32]: 😀😀

**Review by geniuscodes — COMMENTED** [review]
no need to push target folder

File: shipmentNotifier/src/main/java/io/billie/shipmentnotifier/api/shipment/ShipmentResource.java (hunk 1 above)
Comment (mehdiqanbarzadeh) [reply shipmentNotifier/src/main/java/io/billie/shipmentnotifier/api/shipment/ShipmentResource.java:36]: thanks for advice

File: shipmentNotifier/src/main/java/io/billie/shipmentnotifier/exception/CustomizedResponseEntityExceptionHandler.java (hunk 2 above)
Comment (mehdiqanbarzadeh) [reply shipmentNotifier/src/main/java/io/billie/shipmentnotifier/exception/CustomizedResponseEntityExceptionHandler.java:77]: It's need more dynamicity

File: shipmentNotifier/src/main/java/io/billie/shipmentnotifier/service/shipment/impl/ShipmentServiceImpl.java (hunk 3 above)
Comment (mehdiqanbarzadeh) [reply shipmentNotifier/src/main/java/io/billie/shipmentnotifier/service/shipment/impl/ShipmentServiceImpl.java:34]: yeap

File: shipmentNotifier/src/main/java/io/billie/shipmentnotifier/service/shipment/mapper/ShipmentServiceMapper.java (hunk 4 above)
Comment (mehdiqanbarzadeh) [reply shipmentNotifier/src/main/java/io/billie/shipmentnotifier/service/shipment/mapper/ShipmentServiceMapper.java:32]: :D

**Pull request closed without merging**